### ISSUE TITLE
21.0-rc2: Extra 3 in release assets.

### ISSUE THREAD

**Issue by botovq** [protocolbuffers/protobuf#10017]
The language-specific release tarballs and zips for [21.0-rc2][1] have an extra `3` version digit at the beginning when compared with [21.0-rc1][2].

For example, [protobuf-cpp-3.3.21.0-rc-2.tar.gz](https://github.com/protocolbuffers/protobuf/releases/download/v21.0-rc2/protobuf-cpp-3.3.21.0-rc-2.tar.gz) used to be [protobuf-cpp-3.21.0-rc-1.tar.gz](https://github.com/protocolbuffers/protobuf/releases/download/v21.0-rc1/protobuf-cpp-3.21.0-rc-1.tar.gz). The same is true for all other languages.

This isn't special to this release, but is also a bit confusing: the tag is `v21.0-rc2` while the assets have an additional hyphen in their name `21.0-rc-2`. It would be great if only one spelling of `rcN` could be used throughout.

[1]: https://github.com/protocolbuffers/protobuf/releases/tag/v21.0-rc2
[2]: https://github.com/protocolbuffers/protobuf/releases/tag/v21.0-rc1

**Comment by deannagarcia** [comment]
Thanks for pointing this out! I will make sure in the final 21.0 release doesn't have the extra number.

I agree the discrepancy between `-rc2` and `-rc-2` is confusing. Right now we are trying to keep the naming consistent with previous releases, but we will be looking in the future toward renaming/rethinking these source artifacts.

While rethinking the source distributions it would be helpful to understand how users are using them. Would you mind explaining what you use these tar files for?

**Comment by botovq** [comment]
Thanks!

We use  the C++ and Python language runtime for packaging in OpenBSD. The C++ runtime has been used since  Protobuf 2.3.0 and has a sizable number of consumers (about 150-160 packages, mostly via Qt). While the Python bindings were added much later in 2018. Python originally used PyPi, but this never had release candidates and always had a bit of a delay, so it was switched to using the release assets directly.

I am aware of a number of people using both C++ and Python runtime for their development. There has never been a need for the other language bindings, and switching to using `protobuf-all` always seemed more trouble than its worth, so we stuck with what is known to work.

**Comment by jfreeland** [comment]
I also noticed this because I'm using asdf-vm.com to pull protoc and:

```
[13:33:09] jfreeland@work : ~  > asdf list all protoc
21.0-rc1
21.0-rc2
21.0
21.1
3.16.0-rc2
3.16.0
...
```

All of the assets starting with 3.21.0 have been missing the `3.` prefix.

**Comment by deannagarcia** [comment]
The leading 3 in release assets was fixed in 21.0 and 21.1.

As for the naming scheme, you are correct in that the new version for all protoc artifacts will not have a major version prefix. However, if you are using language-specific assets, those will have language-specific versions.
